### ISSUE TITLE
Orientation bug + error

### ISSUE THREAD

**Issue by MichalNemec** [seel-channel/video_viewer#65]
Really neat implementation, congratulations there.

I have this in main.dart -> responsive app any time.
```dart

  await SystemChrome.setPreferredOrientations(
    [
      DeviceOrientation.portraitUp,
      DeviceOrientation.landscapeLeft,
      DeviceOrientation.landscapeRight,
    ],
  );
```

implementation of widget
```dart
VideoViewer(
              autoPlay: false,
              onFullscreenFixLandscape: false,
              controller: controller,
              source: {
                'SubRip Text': VideoSource(
                  video: VideoPlayerController.network(
                    'https://www.speechpad.com/proxy/get/marketing/samples/standard-captions-example.mp4',
                  ),
                  subtitle: {
                    'English': VideoViewerSubtitle.network(
                      'https://felipemurguia.com/assets/txt/WEBVTT_English.txt',
                      type: SubtitleType.webvtt,
                    ),
                  },
                )
              },
            ),
```

scenario 1
1) play video
2) enter fullscreen
3) exit fullscreen
4) rotate to landscape (video stops)

scenario 2
1) play video
2) enter fullscreen
3) rotate to landscape (flicker, black screen, pop back, since there is no UI)
`A VideoViewerController was used after being disposed.`

So to summarize:
is there option when app is rotated to keep the video playing?
how to deal with the rotation on fullscreen?

**Comment by MichalNemec** [comment]
after a little bit of digging i found the issue.
```dart
SliverAlignedGrid.count(
        crossAxisCount: HelperResponsive.numberOfColumns(context, 320), // <--- this is the issue
        mainAxisSpacing: 16,
        crossAxisSpacing: 16,
        itemCount: files.length,
        addAutomaticKeepAlives: false,
        addRepaintBoundaries: false,
        itemBuilder: (context, index) {
        //media
        }
)
```

Because it rebuilds (its just bool for mediaquery width), then it:
a) loses position of the video and since its not in view, its killed aka disposed?
b) when rebuild happens, it just wipes it and kills the video

Could someone help me with this ?

**Comment by andes2912** [comment]
> after a little bit of digging i found the issue.
> 
> ```dart
> SliverAlignedGrid.count(
>         crossAxisCount: HelperResponsive.numberOfColumns(context, 320), // <--- this is the issue
>         mainAxisSpacing: 16,
>         crossAxisSpacing: 16,
>         itemCount: files.length,
>         addAutomaticKeepAlives: false,
>         addRepaintBoundaries: false,
>         itemBuilder: (context, index) {
>         //media
>         }
> )
> ```
> 
> Because it rebuilds (its just bool for mediaquery width), then it: a) loses position of the video and since its not in view, its killed aka disposed? b) when rebuild happens, it just wipes it and kills the video
> 
> Could someone help me with this ?

I get same issue, waiting for the magic

**Comment by andes2912** [comment]
> after a little bit of digging i found the issue.
> 
> ```dart
> SliverAlignedGrid.count(
>         crossAxisCount: HelperResponsive.numberOfColumns(context, 320), // <--- this is the issue
>         mainAxisSpacing: 16,
>         crossAxisSpacing: 16,
>         itemCount: files.length,
>         addAutomaticKeepAlives: false,
>         addRepaintBoundaries: false,
>         itemBuilder: (context, index) {
>         //media
>         }
> )
> ```
> 
> Because it rebuilds (its just bool for mediaquery width), then it: a) loses position of the video and since its not in view, its killed aka disposed? b) when rebuild happens, it just wipes it and kills the video
> 
> Could someone help me with this ?

Hey guys, if you used initState or call function on initState please remove, my problem solved with this
